ui/app/AppLayouts/Profile/popups/TokenListPopup.qml

```diff
@@ -0,0 +1,285 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQml.Models 2.15
+import QtQuick.Layouts 1.15
+
+import StatusQ.Core 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Components 0.1
+import StatusQ.Popups.Dialog 0.1
+import StatusQ.Core.Theme 0.1
+
+import utils 1.0
+
+import shared.panels 1.0
+
+StatusDialog {
+    id: root
+
+    /*required*/ property string sourceName
+    /*required*/ property string sourceImage
+    /*required*/ property string sourceUrl
+    /*required*/ property int sourceUpdatedAt
+    /*required*/ property string sourceVersion
+    /*required*/ property int tokensCount
+    /*required*/ property var tokensListModel // Expected roles: name, symbol, image, chainName, explorerUrl
+
+    signal linkClicked(string link)
+
+    QtObject {
+        id: d
+
+        readonly property int symbolColumnWidth: 90
+        readonly property int addressColumnWidth: 106
+        readonly property int externalLinkBtnWidth: 32
+    }
+
+    width: 521 // by design
+    padding: Style.current.padding
+
+    contentItem: StatusListView {
+        id: list
+
+        implicitHeight: contentHeight
+        model: root.tokensListModel
+        header: ColumnLayout {
+            spacing: 20
+            width: list.width
+
+            CustomSourceInfoComponent {
+                Layout.fillWidth: true
+                Layout.margins: Style.current.padding
+            }
+
+            Separator {}
+
+            CustomHeaderDelegate {}
+        }
+        delegate: CustomDelegate {}
+    }
+
+    header: StatusDialogHeader {
+        headline.title: qsTr("%1 Token List").arg(root.sourceName)
+        headline.subtitle: qsTr("%1 tokens").arg(root.tokensCount)
+        actions.closeButton.onClicked: root.close()
+        leftComponent: StatusSmartIdenticon {
+            asset.name: root.sourceImage
+            asset.isImage: !!asset.name
+        }
+    }
+
+    footer: StatusDialogFooter {
+        spacing: Style.current.padding
+        rightButtons: ObjectModel {
+            StatusButton {
+                text: qsTr("Done")
+                type: StatusBaseButton.Type.Normal
+
+                onClicked: close()
+            }
+        }
+    }
+
+    component CustomTextBlock: ColumnLayout {
+        id: textBlock
+
+        property string title
+        property string text
+
+        Layout.fillWidth: true
+
+        StatusBaseText {
+            Layout.fillWidth: true
+
+            text: textBlock.title
+        }
+
+        StatusBaseText {
+            Layout.fillWidth: true
+
+            text: textBlock.text
+            color: Theme.palette.baseColor1
+        }
+    }
+
+    component CustomExternalLinkButton: Rectangle {
```

Comment:
Hmm, why not use a `StatusFlatButton` here?

Reply:
BC I didn't think on it! ;)
However these specifc colors and sizes design is not fitting with any of the types / sizes of `StatusFlatButton`. I'll give it a try.. maybe I can add a `custom` type and `custom` size so then we cover all the possibilities.

Reply:
No need to add custom types, simply adding a new `textHoverColor` property and overriding the _color_ for the _text_ and _spacing_ is enough!

Thank you for mentioning it!!

Reply:
Yeah I had a déjà-vu there :)